attestation-service/attestation-service/src/policy_engine/opa/mod.rs

```diff
@@ -88,82 +81,56 @@ impl PolicyEngine for OPA {
                 hex::encode(hex)
             };
 
-            let policy_go = GoString {
-                p: policy.as_ptr() as *const c_char,
-                n: policy.len() as isize,
-            };
-
-            let reference = serde_json::json!({ "reference": reference_data_map }).to_string();
-
-            let reference_go = GoString {
-                p: reference.as_ptr() as *const c_char,
-                n: reference.len() as isize,
-            };
-
-            let input_go = GoString {
-                p: input.as_ptr() as *const c_char,
-                n: input.len() as isize,
-            };
-
-            // Call the function exported by cgo and process the returned decision
-            let decision_buf: *mut c_char =
-                unsafe { evaluateGo(policy_go, reference_go, input_go) };
-            let decision_str: &CStr = unsafe { CStr::from_ptr(decision_buf) };
-            let policy_res = decision_str.to_str()?.to_string();
-            debug!("OPA Evaluated: {}", policy_res);
-            if policy_res.starts_with("Error::") {
-                bail!("OPA verification failed: {policy_res}");
-            }
+            // Add policy as data
+            engine
+                .add_policy_from_file(policy_file_path)
+                .context("load policy")?;
 
-            // If a clear approval opinion is given in the evaluation report,
-            // the rejection information will be reflected in the evaluation failure return value.
-            let res_kv: Value = serde_json::from_str(&policy_res)?;
+            let reference_data_map = serde_json::to_string(&reference_data_map)?;
+            let reference_data_map =
+                regorus::Value::from_json_str(&format!("{{\"reference\":{reference_data_map}}}"))?;
+            engine
+                .add_data(reference_data_map)
+                .context("load reference data")?;
 
-            // only if there is a field named `allow` in the evaluation report and
-            // it is false, the evaluation fails. Otherwise the evaluation will be
-            // treated as succees.
-            let allow = res_kv
-                .get("allow")
-                .map(|a| a.as_bool())
-                .map(|a| a.unwrap_or(true))
-                .unwrap_or(true);
+            // Add TCB claims as input
+            engine.set_input_json(&input).context("set input")?;
 
+            let allow = engine.eval_bool_query("data.policy.allow".to_string(), false)?;
```

Comment:
I would prefer supporting policies with arbitrary return value not just true/false. I guess we might be overhauling this to support EAR anyway tho

Reply:
Yes. Maybe we can change this in the PR that supports EAR later. For now the main work is to replace golang with regorus without any logic change, e.g. return value with true/false.